### PR TITLE
Games 1

### PR DESCRIPTION
**Pull request by iy2k22**
Completed GET /api/categories

### CODE REVIEW

File: controllers/index.js
Comment (roisinoreilly) [inline controllers/index.js]: we need to send all responses back to the client on a meaningful key ie. {result}

File: controllers/index.js
Comment (roisinoreilly) [inline controllers/index.js:10]: can you add some error handling to this function

File: models/index.js
Comment (roisinoreilly) [inline models/index.js]: can you refactor this so that if the categories were patched to include more information this would be included in your query

File: __tests__/app.test.js
Comment (roisinoreilly) [inline __tests__/app.test.js:33]: this is a brittle test, it's the equivalent of saying you expect an object to equal another object which isn't robust enough. It works for now with categories because there are no IDs in the data, but won't hold up for future tests. We'd like to see jest matchers .toMatchObject or .objectContaining here please

**Review by roisinoreilly — CHANGES_REQUESTED** [review]
A good start! There's a few changes that need implementing before it can be approved around brittle testing, your query and sending back the response in an object. Once those are done can you send us another PR for this ticket please

File: __tests__/app.test.js
Comment (Poonam-raj) [inline __tests__/app.test.js]: Hey Ibrahim, for any future tests I want you to explicitly check the shape of the data coming back, so checking that properties are there not just checking it matches the unseeded data.

File: __tests__/app.test.js
Comment (Poonam-raj) [inline __tests__/app.test.js:38]: Unfortunately I am going to go backon Roisin's comment about the key that we send data back on - can you send this data back on key of 'categories' so its more meaningful to the user?

File: __tests__/app.test.js
Comment (Poonam-raj) [reply __tests__/app.test.js]: In future please add an assertion checking the length of the returned array too when we're GETting multiple resources

File: app.js
Comment (Poonam-raj) [inline app.js:12]: In your next branch it's worth adding a 500 error handler at the bottom of your app to catch any 'unhandled' errors

File: controllers/index.js
Comment (Poonam-raj) [reply controllers/index.js:10]: What is meant by the above comment is add a 'catch' into your controller to catch and then pass on any rogue errors. There isn't any explicit error handling for this endpoint but we expect a catch block on all controllers to hand errors off to a handler

**Review by Poonam-raj — APPROVED** [review]
Few amendments for yourself, Ibrahim. 

Also some notes on what I want to see in future branches but you don't have to implement for this one (length check for the arr, not using the raw data to check what comes back but actually checking the shape of the data yourself, adding a 500 error handler).

Please read through my comments, make the small amendments I've mentioned (changing the key, adding the catch), then when it's all passing feel free to merge yourself without sending another PR

**Review by Poonam-raj — APPROVED** [review]
Awesome thanks for amending Ibrahim, all good to merge

Future note about the 500 error handler: I don't think you're exporting it out of the error file, you will need to put it into the bottom of your app.js

I want you to take a look at the '4. Refactoring to Individual Error Handling Blocks' section of the error notes (https://notes.northcoders.com/courses/js-back-end/error-handling) at some point as it gives a nice illustration of what we're aiming for with this error file and its usage

**Pull request merged**
